src/fabric.c

```diff
@@ -500,9 +500,15 @@ int DEFAULT_SYMVER_PRE(fi_getinfo)(uint32_t version, const char *node, const cha
 		else
 			tail->next = cur;
 		for (tail = cur; tail->next; tail = tail->next) {
+			if (tail->fabric_attr->prov_name != NULL)
+				FI_DBG(&core_prov, FI_LOG_CORE,
+				       "fi_getinfo: provider name is not NULL\n");
 			tail->fabric_attr->prov_name = strdup(prov->provider->name);
 			tail->fabric_attr->prov_version = prov->provider->version;
 		}
+		if (tail->fabric_attr->prov_name != NULL)
+			FI_DBG(&core_prov, FI_LOG_CORE,
+			       "fi_getinfo: provider name is not NULL\n");
```

Comment:
These aren't in the speed path, so I would switch from FI_DBG to FI_WARN. And the function is already included in the log message.